src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/SunPKCS11.java

```diff
@@ -506,7 +508,9 @@ byte[] exportKey(long hSession, CK_ATTRIBUTE[] attributes, long keyId) throws PK
 
         try {
             long genKeyId = token.p11.C_GenerateKey(wrapKeyGenSession.id(), new CK_MECHANISM(CKM_AES_KEY_GEN), wrapKeyAttributes);
+            isExportWrapKey = true;
             wrapKey = (P11Key)P11Key.secretKey(wrapKeyGenSession, genKeyId, "AES", 256 >> 3, null);
+            isExportWrapKey = false;
```

Comment:
Resetting `isExportWrapKey` needs to happen in the `finally` block.
```java
            isExportWrapKey.set(Boolean.TRUE);
            wrapKey = (P11Key)P11Key.secretKey(wrapKeyGenSession, genKeyId, "AES", 256 >> 3, null);
        } catch (PKCS11Exception e) {
            throw e;
        } finally {
            isExportWrapKey.set(Boolean.FALSE);
```

Reply:
Updated.